### PR TITLE
new build a self-signed jwt guide

### PR DESCRIPTION
**Pull request by susanharper-okta**
## Description:
- **New build a self-signed jwt guide** 
- **Is this PR related to a Monolith release?**NOPE 

### Resolves:

* [OKTA-221719](https://oktainc.atlassian.net/browse/OKTA-221719)

### CODE REVIEW

**Comment by susanharper-okta** [comment]
venkat.viswanathan@okta.com - Please review.

File: packages/@okta/vuepress-site/guides/build-self-signed-jwt/overview/index.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+title: Overview
+---
+JSON Web Tokens (JWTs) allow claims, such as user data, to be represented in a secure manner, helping to ensure trust and security in your app. JWTs are an open standard and there are various libraries available that allow you to create, verify, and inspect them.
+
```
Comment (nbarbettini) [inline packages/@okta/vuepress-site/guides/build-self-signed-jwt/overview/index.md]: We need a little more explanation of the "big picture" here ("use this guide when..."). Feel free to wordsmith this:


---

Okta uses JWTs for many purposes. One purpose is to add an optional additional layer of security to requests your code makes to Okta's [OpenID Connect](https://developer.okta.com/docs/api/resources/oidc/) endpoints that require [client authentication](https://developer.okta.com/docs/api/resources/oidc/#client-authentication-methods), like [/token](https://developer.okta.com/docs/api/resources/oidc/#token).

Follow this guide to understand:
* How to build a signed JWT to use when making requests to these endpoints
* Which claims are required in your JWT payload
* How to sign the JWT with a shared secret or private key

File: packages/@okta/vuepress-site/guides/build-self-signed-jwt/index.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+title: Build a Self-Signed JWT
```
Comment (nbarbettini) [inline packages/@okta/vuepress-site/guides/build-self-signed-jwt/index.md]: How aboutBuild a JWT for Client Authentication? In the future there may be other unrelated reasons to build a signed JWT.

**Comment by nbarbettini** [comment]
Waiting on JS code example, then 🚢

File: packages/@okta/vuepress-site/guides/build-self-signed-jwt/jwt-with-private-key/index.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
+title: Build a JWT With a Private Key
+---
+If you configured your client to use the `private_key_jwt` client authentication method, you must provide the `client_id` in a JWT that you sign with your private key using an RSA or ECDSA algorithm (RS256, RS384, RS512, ES256, ES384, ES512).
```
Comment (venkatviswanathan-okta) [inline packages/@okta/vuepress-site/guides/build-self-signed-jwt/jwt-with-private-key/index.md]:
```suggestion
If you configured your client to use the `private_key_jwt` client authentication method, then you want to build a JWT that you sign with your private key using an RSA or ECDSA algorithm (RS256, RS384, RS512, ES256, ES384, ES512). 
```

Comment (venkatviswanathan-okta) [reply packages/@okta/vuepress-site/guides/build-self-signed-jwt/jwt-with-private-key/index.md]: `client_id` is required as `sub`  and `iss` in all cases.

File: packages/@okta/vuepress-site/guides/build-self-signed-jwt/overview/index.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+* Which claims are required in your JWT payload
+* How to sign the JWT with a shared secret or a private key
+
+### JWT Types
+There are two types of self-signed JWT assertions that you can build for use when you make requests to endpoints that require client authentication:
+
+* JWT With a Shared Key (`client_secret_jwt`)
+* JWT With a Private Key (`private_key_jwt`)
+
+The difference between building these two types of assertions is the algorithm and method used to sign the JWT.
```
Comment (venkatviswanathan-okta) [inline packages/@okta/vuepress-site/guides/build-self-signed-jwt/overview/index.md]:
```suggestion
The difference between building these two types of assertions is the algorithm and key used to sign the JWT.
```

**Pull request merged**